src/doc/trpl/syntax-index.md

```diff
@@ -172,6 +174,47 @@
 * `[…]`: array literal.  See [Primitive Types (Arrays)].
 * `[expr; len]`: array literal containing `len` copies of `expr`.  See [Primitive Types (Arrays)].
 * `[type; len]`: array type containing `len` instances of `type`.  See [Primitive Types (Arrays)].
+* `expr[expr]`: collection indexing.  Overloadable (`Index`, `IndexMut`).
+* `expr[..]`, `expr[a..]`, `expr[..b]`, `expr[a..b]`: collection indexing pretending to be collection slicing, using `Range`, `RangeFrom`, `RangeTo`, `RangeFull` as the "index".
+
+## Deprecated and Obsolete
+
+Additional information on the following may be found in the [Rust release notes], or by searching for relevant RFCs/issues/pull requests.
+
+<!-- 1.0 -->
+
+* `extern crate "foo" as bar` (removed 1.0): syntax for `extern crate foo as bar`.
+* `|:…|`, `|&:…|`, `|&mut:…|` (removed 1.0): syntax for explicit closure kinds (now inferred).
+* `int`, `uint` (removed 1.0): names for `usize` and `isize`.
+* `#[deriving(…)]` (removed 1.0): syntax for `#[derive(…)]`.
+* `use ident::{mod, …}` (removed 1.0): syntax for `use ident::{self, …}`.
+* `box expr` (feature-gated 1.0): unstable syntax for `Box::new(expr)`.
+* `proc(…) …` (removed 1.0): syntax for `move |…| …`.
+
+<!-- 0.12 -->
+
+* `[…, ident.., …]` (feature-gated 0.12): unstable syntax for sub-slice pattern matches.
+* `extern crate bar = foo` (removed 0.12): syntax for `extern crate foo as bar`.
+* `use bar = foo` (removed 0.12): syntax for `use foo as bar`.
+* `<Sized? ident>` (removed 0.12): syntax for `<ident: ?Sized>`.
+* `Gc<type>` (removed 0.12): smart-pointer type; replaced by `Rc<type>` and `Arc<type>`.
```

Comment:
I'm on the fence about `Gc`; the intent is for people looking at old code to figure out what it means.  Since `Gc` was completely removed, they can't rely on deprecation warnings or the like.  But, as you say, it's possible that it might be reintroduced somehow.